sensor/common/deduper/unchanged.go

```diff
@@ -0,0 +1,45 @@
+package deduper
+
+import (
+	"sync/atomic"
+
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// ClosableSet is a concurrency-safe set of strings. Items can be added to the set as long as the set is open.
+// Closing the set ignores all future `Add` operations but allows to fetch already stored items.
+type ClosableSet struct {
```

Comment:
I'm confused by this a bit. Why would you ever add anything to a closed set? Doesn't that imply a timing bug where a synced resource is trying to be added after close?

Reply:
>  Why would you ever add anything to a closed set?

We shouldn't ever. That's why the set is being closed after the reconciliation finishes.

> Doesn't that imply a timing bug where a synced resource is trying to be added after close?

It does, because the set should be closed as soon as sensor observes a `ResourcesSynced` event. If any sensor event gets triggered with `ResourceAction_SYNC_RESOURCE` it's a timing issue.

But my decision was to ignore that event rather than log an error. I will change this a bit to show an error if this case happens.

Reply:
I've changed the code and ended up removing this struct all together. I think it's simpler now.